core_impl/src/test/java/io/opencensus/stats/StatsTestUtil.java

```diff
@@ -53,108 +71,226 @@ private static StatsContextImpl createContext(
+          public Void apply(MeanData arg) {
+            assertThat(arg.getMean()).isWithin(tolerance).of(mean);
+            return null;
+          }
+        },
+        new Function<StdDevData, Void>() {
+          @Override
+          public Void apply(StdDevData arg) {
+            assertThat(arg.getStdDev()).isWithin(tolerance).of(stdDev);
+            return null;
+          }
+        },
+        Functions.<Void>throwIllegalArgumentException());
   }
 
-  private static void assertDistributionAggregateValuesEquivalent(
-      String msg, double tolerance, DistributionAggregate agg1, DistributionAggregate agg2) {
-    Truth.assertWithMessage(msg + " count").that(agg1.getCount()).isEqualTo(agg2.getCount());
-    Truth.assertWithMessage(msg + " mean")
-        .that(agg1.getMean())
-        .isWithin(tolerance)
-        .of(agg2.getMean());
-    Truth.assertWithMessage(msg + " sum").that(agg1.getSum()).isWithin(tolerance).of(agg2.getSum());
-    Truth.assertWithMessage(msg + " range").that(agg1.getRange()).isEqualTo(agg2.getRange());
+  /**
+   * Compare the given {@code WindowData} with expected start and/or end {@code Timestamp}s.
+   *
+   * @param windowData the actual {@code WindowData}.
+   * @param start the expected start {@code Timestamp}. Should be non-null for {@link
+   * CumulativeData}, null for {@link IntervalData}.
+   * @param end the expected end {@code Timestamp}.
+   */
+  static void assertWindowDataEquals(
+      WindowData windowData, @Nullable final Timestamp start, final Timestamp end) {
+    windowData.match(
+        new Function<CumulativeData, Void>() {
+          @Override
+          public Void apply(CumulativeData windowData) {
+            if (start == null) {
+              fail("expected an IntervalData.");
+            }
+            assertThat(windowData.getStart()).isEqualTo(start);
+            assertThat(windowData.getEnd()).isEqualTo(end);
+            return null;
+          }
+        },
+        new Function<IntervalData, Void>() {
+          @Override
+          public Void apply(IntervalData windowData) {
+            if (start != null) {
+              fail("expected a CumulativeData.");
+            }
+            assertThat(windowData.getEnd()).isEqualTo(end);
+            return null;
+          }
+        },
+        Functions.<Void>throwIllegalArgumentException());
+  }
 
-    if (agg1.getBucketCounts() == null && agg2.getBucketCounts() == null) {
-      return;
-    } else {
-      Truth.assertWithMessage(msg + " bucket counts")
-          .that(removeTrailingZeros(agg1.getBucketCounts()))
-          .isEqualTo(removeTrailingZeros(agg2.getBucketCounts()));
+  /**
+   * Remove trailing zeros and return a boxed list of {@code Long}.
+   *
+   * @param longs the input array of primitive type long.
+   * @return a list of boxed object Long, with trailing zeros removed.
+   */
+  static List<Long> removeTrailingZeros(long... longs) {
+    if (longs == null) {
+      return null;
+    }
```

Comment:
It does need to, since I need this method to check cases when one of (or both) the actual and expected bucket counts is null: https://github.com/census-instrumentation/opencensus-java/pull/489/commits/a32e34d2ab00e0cec9486a16b418e64c6436398a#diff-c24b66b69d113e9ee0e6ddcaea2cf739R209